### PR TITLE
feat: improve merge editor result title

### PR DESCRIPTION
**Pull request by Ricbet**
### Types

- [x] 💄 Style Changes

### <samp>🤖 Generated by Copilot at ca2adc1</samp>

* Import modules for memoization and URI manipulation from React and @opensumi/ide-core-browser ([link](https://github.com/opensumi/core/pull/2835/files?diff=unified&w=0#diff-167730c4e8c47ae93d43b381e2817972ab7a70360b88607f65291fbf60d1b926L1-R3))
* Import interface for workspace service from @opensumi/ide-workspace ([link](https://github.com/opensumi/core/pull/2835/files?diff=unified&w=0#diff-167730c4e8c47ae93d43b381e2817972ab7a70360b88607f65291fbf60d1b926R10))
* Create workspace service instance and memoized function to convert absolute URI to relative path based on workspace root ([link](https://github.com/opensumi/core/pull/2835/files?diff=unified&w=0#diff-167730c4e8c47ae93d43b381e2817972ab7a70360b88607f65291fbf60d1b926L19-R35))
* Apply relative path function to output URI when creating result view input data in `grid.tsx` ([link](https://github.com/opensumi/core/pull/2835/files?diff=unified&w=0#diff-167730c4e8c47ae93d43b381e2817972ab7a70360b88607f65291fbf60d1b926L34-R51))

result 视图 title 展示 fsPath 而不是整个 uri

### <samp>🤖 Generated by Copilot at ca2adc1</samp>

Added a feature to display relative paths of output files in the merge editor. Modified `grid.tsx` to import modules and define a function for this feature.

### CODE REVIEW

File: packages/monaco/src/browser/contrib/merge-editor/view/grid.tsx [hunk 1]
```diff
@@ -31,7 +46,9 @@ const TitleHead: React.FC<{ contrastType: EditorViewType }> = ({ contrastType })
       } else if (contrastType === EditorViewType.INCOMING) {
         setHead(input2.getRaw());
       } else if (contrastType === EditorViewType.RESULT) {
-        setHead(new MergeEditorInputData(output.uri, 'Result', output.uri.toString(), '').getRaw());
+        setHead(
+          new MergeEditorInputData(output.uri, 'Result', toRelativePath(output.uri).path.toString(), '').getRaw(),
```
Comment (bytemain) [inline packages/monaco/src/browser/contrib/merge-editor/view/grid.tsx:50]: toRelativePath 实现可以简化点，返回值没必要先转为 URI 再取 path? `rootUri.relative(uri)` 的结果看起来就是我想要的东西

**Review by bytemain — CHANGES_REQUESTED** [review]

File: packages/monaco/src/browser/contrib/merge-editor/view/grid.tsx (hunk 1 above)
Comment (Ricbet) [reply packages/monaco/src/browser/contrib/merge-editor/view/grid.tsx]: done

File: packages/monaco/src/browser/contrib/merge-editor/view/grid.tsx [hunk 2]
```diff
@@ -31,7 +48,7 @@ const TitleHead: React.FC<{ contrastType: EditorViewType }> = ({ contrastType })
       } else if (contrastType === EditorViewType.INCOMING) {
         setHead(input2.getRaw());
       } else if (contrastType === EditorViewType.RESULT) {
-        setHead(new MergeEditorInputData(output.uri, 'Result', output.uri.toString(), '').getRaw());
+        setHead(new MergeEditorInputData(output.uri, 'Result', toRelativePath(output.uri), '').getRaw());
```
Comment (bytemain) [inline packages/monaco/src/browser/contrib/merge-editor/view/grid.tsx:51]: 但是这样不就是就有一个 file:// 的前缀了嘛？

**Review by bytemain — CHANGES_REQUESTED** [review]

File: packages/monaco/src/browser/contrib/merge-editor/view/grid.tsx (hunk 2 above)
Comment (Ricbet) [reply packages/monaco/src/browser/contrib/merge-editor/view/grid.tsx:51]: 不会，最终效果是这样
image

**Review by bytemain — APPROVED** [review]

**Pull request merged**